Core/src/Managers/AudioManager.cpp

```diff
@@ -36,68 +50,35 @@ FMOD::Sound* AudioManager::LoadAudio(const std::string& filePath) {
     return newSound;
 }
 
-void AudioManager::FixedUpdate() {}
-
-void AudioManager::PlayStaticSound(FMOD::Sound* sound, bool isLooping, const glm::vec3& position) {
-    PlaySound(sound, isLooping, position, staticChannel);
-}
-
-void AudioManager::PlayDynamicSound(FMOD::Sound* sound, bool isLooping, const glm::vec3& position) {
-    PlaySound(sound, isLooping, position, dynamicChannel);
-}
-
-void AudioManager::PlaySound(FMOD::Sound* sound, bool isLooping, const glm::vec3& position, FMOD::Channel* channel) {
-    
-    if (!sound) return;
-    if (isLooping)
-        sound->setMode(FMOD_LOOP_NORMAL);
-
-    FMOD_RESULT result = fmodSystem->playSound(sound, nullptr, false, &channel);
-    if (result == FMOD_OK && channel) {
-        channel->setLoopCount(-1);
-        channel->setVolume(1.0f);
-        FMOD_VECTOR audioPosition = GetFMODVector(soundPosition);
-        channel->set3DAttributes(&audioPosition, nullptr);
-        channel->setPaused(false);
-    }
-}
-
-
-// void AudioManager::UpdateChannelPosition(FMOD::Channel* channel, const glm::vec3& position) {
-//     if (!channel) return;
-//
-//     FMOD_VECTOR fmodPosition = GetFMODVector(position);
-//     channel->set3DAttributes(&fmodPosition, nullptr);
-// }
+void AudioManager::FixedUpdate() {} // No use right now, just used for virtual override.
 
 FMOD_VECTOR AudioManager::GetFMODVector(const glm::vec3& position) {
     return { position.x, position.y, position.z };
 }
 
-// TODO: Positioning of Camera , glm::vec3 cameraPosition, glm::vec3& carPosition
 void AudioManager::Update(float deltaTime) {
-    fmodSystem->update();
-    
+    std::vector<UUIDv4::UUID> audioEntitiesUUIDs = EntityManager::getInstance().findEntitiesByComponent(ComponentType::Audio);
+    for (UUIDv4::UUID audioEntityUUID : audioEntitiesUUIDs) {
+        Entity* audioEntity = EntityManager::getInstance().getEntity(audioEntityUUID);
+        AudioComponent* audioComponent = dynamic_cast<AudioComponent*>(
+            audioEntity->getComponent(ComponentType::Audio)
+        );
+        audioComponent->UpdateAttributes(deltaTime);
+
+    }
+    std::vector<UUIDv4::UUID> listenerEntitiesUUIDs = EntityManager::getInstance().findEntitiesByComponent(ComponentType::Listener);
```

Comment:
This is why I don't think listenercomponent should be a singleton, you can already get it through the entity manager.

Not only that but right after you do that, you hae a for loop for listener entities, which there should only be one according to the singleton pattern anyway, and then you get instance immediately after, which meant you didnt need to do the original findEntitiesByComponent anyway!

Reply:
So we should just get the instance of the listener, then update it. Good idea.

But what should we do if we don't enforce singleton behaviour for the listener?

Reply:
????? Look at the code you wrote for this loop, you already get the audio listeners for the for loop, act on them inside the for loop without using the extraneous singleton behaviour